plugin/pkg/replacer/replacer.go

```diff
@@ -58,6 +62,8 @@ func New(r *dns.Msg, rr *dnstest.Recorder, emptyValue string) Replacer {
 		rep.replacements["{duration}"] = strconv.FormatFloat(time.Since(rr.Start).Seconds(), 'f', -1, 64) + "s"
 		if rr.Msg != nil {
 			rep.replacements[headerReplacer+"rflags}"] = flagsToString(rr.Msg.MsgHdr)
+			rep.replacements["{A}"] = answersCount(rr.Msg, dns.TypeA)
```

Comment:
I am thinking that if we move to use metadata instead of the replacement ... then we will have the lazy-evaluation and we can add any metadata reference we want.
it will simplify also dependency between Log and any other plugin.

Reply:
If we want to keep lazy-initialization, we cannot work like a Replacer (that is trying to replace any possible label), we need to parse and query only the label that are in the format.
We need a parsing/checking at setup.

Reply:
I am wrong in former comment. 
We can keep the Replacer and have Lazy-initialization with Metadata. And also add lazy initialization for existing "variables"